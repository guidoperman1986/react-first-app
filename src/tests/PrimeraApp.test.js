import { PrimeraApp } from '../PrimeraApp';
import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
const { render } = require("@testing-library/react");

describe('Pruebas primera app', () => {
    xtest('should show message', () => {
        const saludo = 'Hola, como andas';
        const {getByText} = render(<PrimeraApp saludo={saludo} />);
        expect(getByText(saludo)).toBeInTheDocument();
    })
    
    test('should show PrimeraApp correctly', () => {
        const saludo = 'Hola, como andas';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('should show subtitle', ()=> {
        const saludo = 'Hola, como andas?';
        const subtitulo = 'Soy un subtitulo';

        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo}/>);

        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toEqual('Soy un subtitulo');
    });
    
})
