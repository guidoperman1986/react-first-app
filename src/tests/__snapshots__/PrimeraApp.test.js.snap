// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas primera app should show PrimeraApp correctly 1`] = `
<Fragment>
  <h1>
     
    Hola, como andas
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
